server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexAliasesService.java

```diff
@@ -244,4 +255,43 @@ private static void validateAliasTargetIsNotDSBackingIndex(ClusterState currentS
             );
         }
     }
+
+    /**
+     * A cluster state update task that consists of the cluster state request and the listeners that need to be notified upon completion.
+     */
+    record ApplyAliasesTask(IndicesAliasesClusterStateUpdateRequest request, ActionListener<AcknowledgedResponse> listener)
```

Comment:
We could technically just make this a subclass of `org.elasticsearch.cluster.AckedClusterStateUpdateTask` and call the `execute` method (that would just forward to `applyAliasActions`) that we'd have to implement in `executeTask` and make this way shorter I think?

Reply:
let me check :)

Reply:
I think I'd rather not do this, `AckedClusterStateUpdateTask` is the unbatched one. Maybe we can do a follow-up that introduces a new abstraction here.

Reply:
Sorry about the noise then, makes sense :)

Reply:
I liked that we didn't have to forward to `applyAliasActions` anymore the way the code was, but implementing the all the methods of the listener was tedious. So, I will implement this suggestion.

Reply:
Ah I didn't realize this. I will revert the change.